boot/boot.go

```diff
@@ -429,8 +429,50 @@ func updateManagedBootConfigForBootloader(dev Device, mode string) (updated bool
 		return false, err
 	}
 	// boot config update can lead to a change of kernel command line
-	if err := observeCommandLineUpdate(dev.Model()); err != nil {
+	_, err = observeCommandLineUpdate(dev.Model(), commandLineUpdateReasonSnapd, gadgetSnapOrDir)
+	if err != nil {
 		return false, err
 	}
 	return tbl.UpdateBootConfig()
 }
+
+// CommandLineComponentUpdate handles the update of a component that contributes
+// to the kernel command line of the run system. Returns true when a change in
+// command line has been observed and a reboot is needed. The reboot, if needed,
+// should be requested as the the earliest possible occasion.
+func CommandLineComponentUpdate(dev Device, gadgetSnapOrDir string) (needsReboot bool, err error) {
```

Comment:
this should be called something like

UpdateCommandLineForGadgetComponent

do be consistent with the other function naming here

Reply:
Renamed